tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+                  associated to the upload.
+                  
+=cut
+sub deleteUploadInDatabase {
+	my($dbh, $uploadID, $tarchiveID, $sessionID, $intermediaryFilesRef, $picFilesRef)= @_;
+	
+	$dbh->{'AutoCommit'} = 0;
+	
+	my $query = "DELETE FROM notification_spool WHERE ProcessID = ?";
+	$dbh->do($query, undef, $uploadID);
+	
+	$query = "DELETE FROM tarchive_files WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	$query = "DELETE FROM tarchive_series WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	if(@$intermediaryFilesRef) {
+		my @intermediaryFileIds = map { $_->{'IntermedID'} } @$intermediaryFilesRef;
+		$query = sprintf(
+		    'DELETE FROM files_intermediary WHERE IntermedID IN (%s)', 
+		    join(',', map { '?' } (1..@intermediaryFileIds))
+		);
+		$dbh->do($query, undef, @intermediaryFileIds);
+    }
+    
+ 	if(@$picFilesRef) {
+	    $query  = sprintf(
+	        "DELETE FROM parameter_file WHERE FileID IN (%s)",
+	        join(',', map { '?' } (1..@$picFilesRef))
+	    );
+	    $dbh->do($query, undef, map { $_->{'FileID'} } @$picFilesRef);
+	}
+	   	
+	$query = sprintf(
+	    "DELETE FROM files WHERE FileID IN (%s)",
+	     join(',', map { '?' } (1..@$intermediaryFilesRef))
+	);
+	$dbh->do($query, undef, map { $_->{'FileID'} } @$intermediaryFilesRef);
+	$query = "DELETE FROM files WHERE TarchiveSource = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	$query = "DELETE FROM mri_protocol_violated_scans WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	$query = "DELETE FROM mri_violations_log WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	$query = "DELETE FROM MRICandidateErrors WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	$query = "DELETE FROM mri_upload WHERE UploadID = ?";
+	$dbh->do($query, undef, $uploadID);
+	
+	$query = "DELETE FROM tarchive WHERE TarchiveID = ?";
+	$dbh->do($query, undef, $tarchiveID);
+	
+	# If the upload to delete is the last upload that was part of the 
+	# session associated to it, then set the session's 'Scan_done' flag
+	# to 'N'
+	$query = "SELECT UploadID FROM mri_upload WHERE SessionID = ?";
+	my $rowsRef = $dbh->selectall_arrayref($query, { Slice => {} }, $sessionID);
+	if(@$rowsRef == 0) {
+	    $query = "UPDATE session SET Scan_done = 'N' WHERE ID = ?";
+	    $dbh->do($query, undef, $sessionID);
+	}
+	
+	$dbh->commit;
+    $dbh->{'AutoCommit'} = 0;
+}
+
+=pod
+
+=head3 deleteUploadFiles($archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef)
+
+This method deletes form the file system all the files tied to the upload that were listed in
+tables C<files>, C<files_intermediary> and <parameter_file>, along with the back up of the 
+archive created by the MRI pipeline when the upload was processed. A warning is issued for any
+file that could not be deleted.
+
+INPUTS:
+ 
+  - $archiveLocation: full path of the archive backup created by the MRI pipeline when the upload
```

Comment:
description of archiveLocation confusing here too

Reply:
I changed the description. I hope it's clear now.